specification/securitydevops/resource-manager/Microsoft.SecurityDevOps/preview/2022-09-01-preview/securitydevops.json

```diff
@@ -0,0 +1,2692 @@
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "AzureDevOpsConnector_ListBySubscription": {
+            "$ref": "./examples/AzureDevOpsConnectorListBySubscription.json"
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.SecurityDevOps/operations": {
+      "get": {
+        "description": "Returns list of operations supported by the Microsoft.SecurityDevOps provider.",
+        "operationId": "Operations_List",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Operation details.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/OperationListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "Operations_List": {
+            "$ref": "./examples/OperationsList.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ActionableRemediation": {
+      "type": "object",
+      "properties": {
+        "state": {
+          "$ref": "#/definitions/ActionableRemediationState"
+        },
+        "severityLevels": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "x-ms-identifiers": []
+        },
+        "categories": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RuleCategory"
+          },
+          "x-ms-identifiers": []
+        },
+        "branchConfiguration": {
+          "$ref": "#/definitions/TargetBranchConfiguration"
+        }
+      }
+    },
+    "ActionableRemediationState": {
+      "enum": [
+        "None",
+        "Disabled",
+        "Enabled"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "ActionableRemediationState",
+        "modelAsString": true,
+        "modelAsExtensible": true
+      }
+    },
+    "AuthorizationInfo": {
+      "type": "object",
+      "properties": {
+        "code": {
+          "description": "Gets or sets one-time OAuth code to exchange for refresh and access tokens.",
+          "type": "string",
+          "example": "00000000000000000000.",
+          "x-ms-secret": true
+        }
```

Comment:
x-ms-secret apis are not allowed to be GET-table. They must be retrievable only by POST actions, if at all [sometimes they are effectively write-only]

This is a critically important part of the API to model correctly.

Reply:
this is in the model so that it can set during PUT and we clear it during get. and in general RPaaS doesnt return any property marked as secret back.

Reply:
You should clarify the doc however